engine/runtime/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -122,17 +122,30 @@ class Compiler(
     }
   }
 
-  private def parseModule(module: Module): Unit = {
+  /** Runs part of the compiler to generate docs from Enso code.
+    * @param module - the scope from which docs are generated.
+    */
+  def generateDocs(module: Module): Module = {
+    initializeBuiltinsIr()
+    parseModule(module, isGeneratingDocs = true)
+    module
+  }
+
+  private def parseModule(
+    module: Module,
+    isGeneratingDocs: Boolean = false
+  ): Unit = {
     module.ensureScopeExists()
     module.getScope.reset()
     val moduleContext = ModuleContext(
       module          = module,
       freshNameSupply = Some(freshNameSupply),
       compilerConfig  = config
     )
-    val parsedAST        = parse(module.getSource)
-    val expr             = generateIR(parsedAST)
-    val discoveredModule = recognizeBindings(expr, moduleContext)
+    val parsedAST = parse(module.getSource)
+    val expr      = generateIR(parsedAST)
+    val discoveredModule =
+      recognizeBindings(expr, moduleContext, isGeneratingDocs)
```

Comment:
You are using a wrong code formatter – please setup scala fmt. These lines are properly formatted according to our config.